### ISSUE TITLE
Require defining chat pattern

### ISSUE THREAD

**Issue by wvffle** [PrismarineJS/mineflayer#884]
## Is your feature request related to a problem?
Right now we have kind of 'universal' chat pattern added. Well, in my opinion  it's wrong approach. There are a couple of servers that don't match the pattern and also `chat` pattern matches whispers with quoted chat messages (which is expected<sup><strong>1</strong></sup>)

## Describe the solution you'd like
I think that if programmer would like to use `chat`/`whisper` events then he should add patterns himself first. There should not be a predefined pattern.(Which in some case could not match the chat)

That way programmers will be forced to define custom patterns if they want to handle chat thus gaining more control over it.

We can expose current chat patterns as `mineflayer.chatPattern` and `mineflayer.whisperPattern` for people to use.

```js
cosnt bot = require('./bot')
const handleCommands = require('./commands/handler')


bot.on('chat', handleCommands)
// The handler is added but it's not firing
// The chat pattern for `chat` is not defined yet

// Let's use mineflayer's regex for (universal) chat pattern
const { chatPattern } = require('mineflayer')
bot.chatAddPattern('chat', chatPattern)
// Previously added handler works now
```

## Describe alternatives you've considered
We can also warn/throw if programmer adds handlers for `chat` or `whisper` events and doesn't add chat pattern for respective one in the same event loop tick

## TL;DR
Just require defining chat pattern.

## Additional context

<details>
<summary><strong>Annotation 1</strong></summary>
Let's think about this case:

We have a server, where chat message looks like that: `[ADMIN] wvffle: i like waffles`

It's clear that `[ADMIN]` is rank of an user, which is not desired to catch in `chat` event as username. Current chat pattern skips everything in square brackets.

Now, let's say that the whisper looks like this: `[void0 -> me] [ADMIN] wvffle: i like waffles`

In this case both `chat` and `whisper` will fire. `[void0 -> me]` is treated like any other prefix before username in `chat` event.
</details>

**Comment by rom1504** [comment]
I don't see how that would change the current situation.
There are 2 kind of people :
1. People that can read the doc and are willing to do some minimal effort :
they can already define some pattern themselves
2. People that don't want to do any effort : these people currently just
ask in discord, with the change you are proposing they will still ask in
discord

Plus, we still don't have any automated testing on anything non-vanilla so
it doesn't make too much sense to bring non vanilla support that way.
